src/main/java/com/flipkart/ranger/healthcheck/Healthcheck.java

```diff
@@ -16,6 +16,15 @@
 
 package com.flipkart.ranger.healthcheck;
 
+/**
+ * An interface used to check the health of the service
+ */
 public interface Healthcheck {
+
+    /**
+     * return healthy/unhealthy status of the service
+     *
+     * @return health status
+     */
     public HealthcheckStatus check();
```

Comment:
interfaces with single methods are called functional interfaces or SAM interfaces.  see quick link that I could pull up, https://dzone.com/articles/introduction-functional-1  they are candidates for lambdas so you can do lot of declarative coding isntead of creating scaffolding around these.